### PR TITLE
Supporting documents update

### PR DESCRIPTION
**Pull request by dsamojlenko**
# What this does

Refactors /en/documents to be more inline with the design

## To test

- Create one or more conditions, upload one or more documents for each
- Navigate to `/en/documents` and you'll see all the documents aggregated there, with the checkbox for the conditions they were uploaded to checked
- Go back and edit one or more conditions and upload one or more additional documents
- Navigate to `/en/documents` all documents should be represented
- On `/en/documents`, associate documents to more conditions by checking the boxes
- Save and return to `/en/documents` and checkboxes should be as you left them
- Go back to Edit the conditions you changed, and the changes should be synced back

## Notes/Limitations

- Since we're not working with a relational database, some of this can be a bit buggy. 
- Removing files on either the condition edit or the documents index may not sync properly. 
- Uploading files of the same name may just create duplicates.

### CODE REVIEW

**Comment by sidewalkballet** [comment]
(just noting that https://trello.com/c/o6v6TMEO/538-logic-for-supporting-documents-is-never-marked-as-complete-on-the-dashboard-page still exists)

File: api/utils/DocumentsHelper.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+    documents.forEach((document) => {
+      if (document.conditions.indexOf(conditionId) !== -1) {
+        conditionDocuments.push(document.name);
+      }
+    });
+
+    return conditionDocuments;
+  }
+  return false;
```
Comment (jeffmaher) [inline api/utils/DocumentsHelper.js]: Suggestion: Might be better to return `undefined` or `null` here.

File: api/utils/DocumentsHelper.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
+const castArray = require('./ArrayHelpers').castArray;
+
+const getAllDocuments = (medicalReport) => {
+  if (medicalReport.supportingDocuments) {
+    return medicalReport.supportingDocuments;
+  }
+  return false;
```
Comment (jeffmaher) [inline api/utils/DocumentsHelper.js]: Suggestion: Might be better to return `undefined` or `null` here.

File: api/utils/DocumentsHelper.js (hunk 1 above)
Comment (dsamojlenko) [reply api/utils/DocumentsHelper.js]: agreed

**Pull request merged**